LSP: conda testing can't find lfortran

It must find it in the $PATH, like before. 

```
        server_path = None
        if 'LFORTRAN_PATH' in os.environ:
            server_path = os.environ['LFORTRAN_PATH']
        if server_path is None or not os.path.exists(server_path):
            server_path = Path(__file__).absolute().parent.parent.parent.parent / "src" / "bin" / "lfortran"
        if server_path is None:
            server_path = shutil.which('lfortran')
        if server_path is None:
            raise RuntimeError('cannot determine location of lfortran')
        server_path = Path(server_path)
        if not (server_path.exists() and os.access(server_path, os.X_OK)):
>           raise RuntimeError(f'Invalid or non-executable path to lfortran: {server_path}')
E           RuntimeError: Invalid or non-executable path to lfortran: /home/runner/work/lfortran-conda-testing/lfortran-conda-testing/lfortran/src/bin/lfortran

tests/server/tests/conftest.py:39: RuntimeError
=========================== short test summary info ============================
ERROR tests/server/tests/test_document_manipulation.py::test_document_manipulation - RuntimeError: Invalid or non-executable path to lfortran: /home/runner/work/lfortran-conda-testing/lfortran-conda-testing/lfortran/src/bin/lfortran
ERROR tests/server/tests/test_features.py::test_goto_definition - RuntimeError: Invalid or non-executable path to lfortran: /home/runner/work/lfortran-conda-testing/lfortran-conda-testing/lfortran/src/bin/lfortran
ERROR tests/server/tests/test_features.py::test_diagnostics - RuntimeError: Invalid or non-executable path to lfortran: /home/runner/work/lfortran-conda-testing/lfortran-conda-testing/lfortran/src/bin/lfortran
ERROR tests/server/tests/test_features.py::test_configuration_caching - RuntimeError: Invalid or non-executable path to lfortran: /home/runner/work/lfortran-conda-testing/lfortran-conda-testing/lfortran/src/bin/lfortran
ERROR tests/server/tests/test_features.py::test_rename - RuntimeError: Invalid or non-executable path to lfortran: /home/runner/work/lfortran-conda-testing/lfortran-conda-testing/lfortran/src/bin/lfortran
ERROR tests/server/tests/test_features.py::test_document_highlight - RuntimeError: Invalid or non-executable path to lfortran: /home/runner/work/lfortran-conda-testing/lfortran-conda-testing/lfortran/src/bin/lfortran
============================== 6 errors in 0.10s ===============================
```

--- a/tests/server/tests/conftest.py
+++ b/tests/server/tests/conftest.py
@@ -30,7 +30,7 @@ def client(request: pytest.FixtureRequest) -> Iterator[LFortranLspTestClient]:
         server_path = os.environ['LFORTRAN_PATH']
     if server_path is None or not os.path.exists(server_path):
         server_path = Path(__file__).absolute().parent.parent.parent.parent / "src" / "bin" / "lfortran"
-    if server_path is None:
+    if server_path is None or not os.path.exists(server_path):
         server_path = shutil.which('lfortran')
     if server_path is None:
         raise RuntimeError('cannot determine location of lfortran')

